src/cli/commands/dht/provide.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+module.exports = {
+  command: 'provide <key>',
+
+  describe: 'Announce to the network that you are providing given values.',
+
+  builder: {
+    recursive: {
+      alias: 'r',
+      recursive: 'Recursively provide entire graph.',
+      default: false
+    }
+  },
+
+  handler ({ ipfs, key, resolve }) {
+    // TODO add recursive option
```

Comment:
Please pass this option to `ipfs.dht.provide` so it can throw and inform the user the option is not implemented yet.